src/creosote/parsers.py

```diff
@@ -58,19 +59,10 @@ def pyproject(self, deps_file: str, sections: list):
 
         return sorted(deps)
 
+
     def requirements(self, deps_file: str):
         """Return dependencies from requirements.txt-format file."""
-        deps = []
-        with open(deps_file, "r") as infile:
-            contents = infile.readlines()
-
-        for line in contents:
-            if line.startswith("#") or line.startswith(" "):
-                continue
-            parsed_dep = self.parse_dep_string(line)
-            deps.append(parsed_dep)
-
-        return sorted(deps)
+        return sorted([dep.name for dep in RequirementsFile.from_file(deps_file).requirements])
```

Comment:
This is great, but I see a linting warning here:

```
Argument of type "list[str | None]" cannot be assigned to parameter "__iterable" of type "Iterable[SupportsRichComparisonT@sorted]" in function "sorted"
  "list[str | None]" is incompatible with "Iterable[SupportsRichComparisonT@sorted]"
    TypeVar "_T_co@Iterable" is covariant
      Type "str | None" cannot be assigned to type "SupportsRichComparison"
        Type "str | None" cannot be assigned to type "SupportsRichComparison"
          Type "None" cannot be assigned to type
```


...which would e.g. be solved by only adding `dep.name` to the list if it is not `None`. So I would prefer something like this:

```suggestion
        deps = RequirementsFile.from_file(deps_file).requirements
        return sorted([dep.name for dep in deps if dep.name is not None])
```

Reply:
Thanks! Suggestion added in 312ebc0